pkg/router/controller/route_secret_manager.go

```diff
@@ -0,0 +1,290 @@
+			} else if len(tls.ExternalCertificate.Name) > 0 {
+				errs := validateTLSExternalCertificate(route, fldPath.Child("externalCertificate"), sarc, secrets)
+				result = append(result, errs...)
+			}
+		}
+	//passthrough term should not specify any cert
+	case routev1.TLSTerminationPassthrough:
+		if tls.ExternalCertificate != nil {
+			if len(tls.ExternalCertificate.Name) > 0 {
+				result = append(result, field.Invalid(fldPath.Child("externalCertificate"), tls.ExternalCertificate.Name, "passthrough termination does not support certificates"))
+			}
+		}
+	default:
+		validValues := []string{string(routev1.TLSTerminationEdge), string(routev1.TLSTerminationPassthrough), string(routev1.TLSTerminationReencrypt)}
+		result = append(result, field.NotSupported(fldPath.Child("termination"), tls.Termination, validValues))
+	}
+
+	return result
+}
+
+// validateTLSExternalCertificate tests different pre-conditions required for
+// using externalCertificate. Called by validateTLS.
+func validateTLSExternalCertificate(route *routev1.Route, fldPath *field.Path, sarc authorizationclient.SubjectAccessReviewInterface, secretsGetter corev1client.SecretsGetter) field.ErrorList {
+	tls := route.Spec.TLS
+
+	errs := field.ErrorList{}
+	// The router serviceaccount must have permission to get/list/watch the referenced secret.
+	// The role and rolebinding to provide this access must be provided by the user.
+	if err := authorizationutil.Authorize(sarc, &user.DefaultInfo{Name: routerServiceAccount},
+		&authorizationv1.ResourceAttributes{
+			Namespace: route.Namespace,
+			Verb:      "get",
+			Resource:  "secrets",
+			Name:      tls.ExternalCertificate.Name,
+		}); err != nil {
+		errs = append(errs, field.Forbidden(fldPath, "router serviceaccount does not have permission to get this secret"))
+	}
+
+	if err := authorizationutil.Authorize(sarc, &user.DefaultInfo{Name: routerServiceAccount},
+		&authorizationv1.ResourceAttributes{
+			Namespace: route.Namespace,
+			Verb:      "watch",
+			Resource:  "secrets",
+			Name:      tls.ExternalCertificate.Name,
+		}); err != nil {
+		errs = append(errs, field.Forbidden(fldPath, "router serviceaccount does not have permission to watch this secret"))
+	}
+
+	if err := authorizationutil.Authorize(sarc, &user.DefaultInfo{Name: routerServiceAccount},
+		&authorizationv1.ResourceAttributes{
+			Namespace: route.Namespace,
+			Verb:      "list",
+			Resource:  "secrets",
+			Name:      tls.ExternalCertificate.Name,
+		}); err != nil {
+		errs = append(errs, field.Forbidden(fldPath, "router serviceaccount does not have permission to list this secret"))
+	}
+
+	// The secret should be in the same namespace as that of the route.
+	secret, err := secretsGetter.Secrets(route.Namespace).Get(context.TODO(), tls.ExternalCertificate.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return append(errs, field.NotFound(fldPath, err))
```

Comment:
```suggestion
			return append(errs, field.NotFound(fldPath, err.Error()))
```
Using `err.Error()` results in error messages like this:
```text
spec.tls.externalCertificate: Not found: "secrets \"tls-secret\" not found"
```
instead of this:
```text
spec.tls.externalCertificate: Not found: errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"secrets \"tls-secret\" not found", Reason:"NotFound", Details:(*v1.StatusDetails)(0xc0001a8c60), Code:404}}
```
(Thanks, @lihongan, for pointing out the overly verbose error message!)

Reply:
Thanks for the suggestion, updated to `err.Error()`